pkg/cloudprovider/drift.go

```diff
@@ -40,11 +40,57 @@ import (
 )
 
 const (
+	NodeClassDrift    cloudprovider.DriftReason = "NodeClassDrift"
 	K8sVersionDrift   cloudprovider.DriftReason = "K8sVersionDrift"
 	ImageVersionDrift cloudprovider.DriftReason = "ImageVersionDrift"
 	SubnetDrift       cloudprovider.DriftReason = "SubnetDrift"
 )
 
+func (c *CloudProvider) isNodeClassDrifted(ctx context.Context, nodeClaim *karpv1.NodeClaim, nodeClass *v1alpha2.AKSNodeClass) (cloudprovider.DriftReason, error) {
+	// First check if the node class is statically drifted to save on API calls.
+	if drifted := c.areStaticFieldsDrifted(nodeClaim, nodeClass); drifted != "" {
```

Comment:
nit: prefer naming to be more detailed than `drifted`.

maybe `staticFieldsDrifted`